storage/engine/mvcc_test.go

```diff
@@ -2130,3 +2130,32 @@ func TestMVCCGarbageCollectIntent(t *testing.T) {
 		t.Fatal("expected error garbage collecting an intent")
 	}
 }
+
+// TestResovleIntentWithLowerEpoch test a intent with higher epoch
```

Comment:
+// TestResovleIntentWithLowerEpoch verifies that trying to resolve
+// an intent at an epoch that is lower than the epoch of the intent
+// leaves the intent untouched.